### PR TITLE
Document qabel-desktop REST-server

### PR DESCRIPTION
**Pull request by audax**
Documentation for the qabel-desktop REST-server

see: #99 and https://github.com/Qabel/qabel-desktop/issues/26

Should be ready to merge yet.

### CODE REVIEW

File: _docs/Qabel-Desktop-REST.md [hunk 1]
```diff
@@ -0,0 +1,177 @@
...
+
+## Drops
+
+### Retrieve Drops
+
+* HTTP-Method: GET
+* Ressource: `/drop`
+
+Returns a list of all drop messages from all configured DropURLs.
+See: [Client Drop](../Qabel-Client-Drop]
```
Comment (jan-schreib) [inline _docs/Qabel-Desktop-REST.md]: Something went wrong with the URL there.

File: _docs/Qabel-Desktop-REST.md [hunk 2]
```diff
@@ -0,0 +1,177 @@
...
+|:----------------:|------|
+| 200 | Drops sent |
+| 204 | No new drops available |
+
+### Check for drops
+
+* HTTP-Method: HEAD
+* Ressource: `/drop`
+
+Same as retrieving drops, but without sending them.
```
Comment (jan-schreib) [inline _docs/Qabel-Desktop-REST.md]: Retrieving drops also sends them? Or do I missunderstand the meaning of "send" here?

File: _docs/Qabel-Desktop-REST.md [hunk 3]
```diff
@@ -0,0 +1,177 @@
...
+| key | Key to update, can be in the form of foo.bar.baz |
+| value | Value to set for the key |
+
+#### Return values
+
+|HTTP status code|reason|
+|:----------------:|------|
+| 200 | Setting updated |
+| 400 | value is not valid for this key |
+| 405 | key is not valid |
```
Comment (jan-schreib) [inline _docs/Qabel-Desktop-REST.md]: You are using uppercase most of the time, please do it everywhere.

File: _docs/Qabel-Desktop-REST.md [hunk 4]
```diff
@@ -0,0 +1,177 @@
...
```
Comment (jan-schreib) [inline _docs/Qabel-Desktop-REST.md]: Upper/Lowercase issue here, too.

File: _docs/Qabel-Desktop-REST.md [hunk 5]
```diff
@@ -0,0 +1,177 @@
...
+| --- | ----------- |
+| settings | JSON document of the settings |
+
+#### Return values
+
+|HTTP status code|reason|
+|:----------------:|------|
+| 200 | Setting updated |
+| 400 | Settings contain invalid keys or values |
+| 405 | Settings is not a valid JSON document |
```
Comment (jan-schreib) [inline _docs/Qabel-Desktop-REST.md]: Setting or settings?

File: _docs/Qabel-Desktop-REST.md [hunk 6]
```diff
@@ -0,0 +1,177 @@
+---
+title: Qabel Desktop REST Server
```
Comment (jan-schreib) [inline _docs/Qabel-Desktop-REST.md]: Looks weird in the rendered document.

**Comment by jan-schreib** [comment]
Again, I should read the whole comment before starting to work.

**Comment by audax** [comment]
That should do it.

File: _docs/Qabel-Desktop-REST.md [hunk 7]
```diff
@@ -0,0 +1,177 @@
+---
+title: Local REST Server
+---
+
+# REST Server
+
+## Abstract
+
+A local webserver which serves as a wrapper around the Qabel-core and provides an API that other applications can use. This provides freedom of choice in regards to the GUI or CLI implementations and allows the local server to run in the background where it can provide a permament process for modules like messaging and file syncing.
```
Comment (jasinai) [inline _docs/Qabel-Desktop-REST.md]: permanent, not permament :)

**Comment by audax** [comment]
Plans to implement the REST server are postponed.

See: https://github.com/Qabel/qabel-desktop/pull/27

**Pull request closed without merging**